src/components/Home/actions/getLaunches.ts

```diff
@@ -5,17 +5,20 @@ import {
 } from './launchActionCreators';
 import { Launches } from '../constants/launches';
 import { fetchLaunches } from '../services/fetchLaunches';
+import { fetchDefaults } from '../../../shared/api/constants/fetchDefaults';
 import type { Dispatch } from '@reduxjs/toolkit';
 import type { LaunchAction } from '../types/LaunchActions';
 import type { LaunchResponse } from '../types/LaunchResponse';
 
 export const getLaunches =
-  (filterValue: Launches) =>
+  (filterValue: Launches, page: number = fetchDefaults.FIRST_PAGE) =>
   async (dispatch: Dispatch<LaunchAction>): Promise<void> => {
-    dispatch(launchRequestAction(filterValue));
+    const isScrollLoading: boolean = page !== 1;
+    dispatch(launchRequestAction(filterValue, isScrollLoading));
     try {
-      const response: LaunchResponse = await fetchLaunches(filterValue);
-      dispatch(launchSuccessAction(filterValue, response.data.results));
+      const response: LaunchResponse = await fetchLaunches(filterValue, page);
+      const { results, count }: LaunchResponse['data'] = response.data;
```

Comment:
Providing type like this not a good idea

Reply:
LaunchResponse['data'] changed to new type `LaunchResponseData`